src/main/java/com/redhat/devtools/intellij/knative/actions/func/UndeployAction.java

```diff
@@ -0,0 +1,45 @@
+package com.redhat.devtools.intellij.knative.actions.func;
+
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.ui.Messages;
+import com.redhat.devtools.intellij.common.utils.UIHelper;
+import com.redhat.devtools.intellij.knative.actions.DeleteAction;
+import com.redhat.devtools.intellij.knative.kn.Kn;
+import com.redhat.devtools.intellij.knative.tree.KnFunctionNode;
+import com.redhat.devtools.intellij.knative.tree.ParentableNode;
+import com.redhat.devtools.intellij.knative.utils.TreeHelper;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class UndeployAction extends DeleteAction {
+    public UndeployAction() {
+        super(true, KnFunctionNode.class);
+    }
+
+    @Override
+    protected String getActionName(boolean firstLetterCapital) {
+        String u = firstLetterCapital ? "U" : "u";
+        return u + "ndeploy";
+    }
+
+    @Override
+    protected void doDelete(Project project, Kn kncli, Class type, Map<Class, List<ParentableNode>> resourcesByClass) throws IOException {
+        deleteResources(type, resourcesByClass, kncli);
+        TreeHelper.refreshFuncTree(project);
```

Comment:
I'm still having issue with the button label:

![image](https://user-images.githubusercontent.com/695993/149754337-482b81e0-dadb-485a-bdbc-a77424fd2542.png)

Reply:
My bad, only focussed on main message. Fixed